optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/optional/score/FlightCrewSchedulingConstraintProvider.java

```diff
@@ -35,8 +35,9 @@ private Constraint requiredSkill(ConstraintFactory constraintFactory) {
     }
 
     private Constraint flightConflict(ConstraintFactory constraintFactory) {
-        return constraintFactory.fromUniquePair(FlightAssignment.class, Joiners.equal(FlightAssignment::getEmployee))
-                .filter((first, second) -> second.getFlight().overlaps(first.getFlight()))
+        return constraintFactory.fromUniquePair(FlightAssignment.class, Joiners.equal(FlightAssignment::getEmployee),
+                Joiners.overlaps(fa -> fa.getFlight().getDepartureUTCDateTime(),
+                        fa -> fa.getFlight().getArrivalUTCDateTime()))
```

Comment:
Great to see this!

Small suggestion: revmove `Joiners.` for consistency because the other ConstraintProviders seem to do a static import of all Joiners methods.